### PR TITLE
Listener config

### PR DESCRIPTION
**Pull request by ilackarms**
This PR contains two sets of changes -
1. refactor (and simplification) of the translator
2. add Listeners to Roles

Listeners work as follows:
* Listeners explicitly reference the VirtualServices (routes) that they'll route to. 
* The Listener filter chain will contain the HttpConnectionManager filter as long as it references >= 1 virtual service.
* Listeners have a `map<string, interface{}>` field called Config where extra listener config can live (for example, other Network filters). this is where our custom TCP filters will get configuration
*  Listener specifies the bind address and Port to bind on.
* the "Default Role" (ingress/gateway) is created at boot time by Gloo (unless the user provides one with the same name) with an "insecure" listener on 8080 and a "secure" listener on 8443 (this is basically what we have now, just modeled for the new API)

@yuval-k  @talnordan ptal :heart:

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -41,4 +41,4 @@ jobs:
             # Nomand can't run on docker ci
             # RUN_NOMAD_TESTS: "1"
             RUN_KUBE_TESTS: "1"
-            ENVOY_IMAGE_TAG: "0.2.28"
\ No newline at end of file
+            ENVOY_IMAGE_TAG: "0.3.0"
```
Comment (talnordan) [inline .circleci/config.yml]: Nit: No newline at end of file.

File: docs/installation/docker.md [hunk 2]
```diff
@@ -3,7 +3,7 @@
 ## Prerequisite
 
  1. [Docker](https://www.docker.com/)
- 2. [glooctl](https://github.com/solo-io/glooctl) v0.2.27 or above.
+ 2. [glooctl](https://github.com/solo-io/glooctl) v0.3.0 or above.
```
[the remark this answers is not in the document]
Comment (ilackarms) [reply docs/installation/docker.md]: we don't have a  v0.2.27 either :crying_cat_face:

File: internal/control-plane/translator/listener.go [hunk 3]
```diff
@@ -0,0 +1,93 @@
...
+	"github.com/solo-io/gloo/pkg/api/types/v1"
+	"github.com/solo-io/gloo/pkg/log"
+)
+
+func (t *Translator) computeListener(listener *v1.Listener, inputs *snapshot.Cache) *envoyapi.Listener {
+	rdsName := routeConfigName(listener)
+	httpConnMgr := t.computeHttpConnectionManager(rdsName)
+
+	// TODO (ilacakrms): add more network filters here
+	networkFilters := []envoylistener.Filter{httpConnMgr}
```
[the remark this answers is not in the document]
Comment (ilackarms) [reply internal/control-plane/translator/listener.go]: good call

File: install/docker-compose/README.md [hunk 4]
```diff
@@ -2,7 +2,7 @@
 
 ## Using glooctl
 
- 1. Make sure you have version 0.2.27 or above of `glooctl` installed.
+ 1. Make sure you have version 0.3.0 or above of `glooctl` installed.
```
[the remark this answers is not in the document]
Comment (ilackarms) [reply install/docker-compose/README.md]: i created a new release for 0.3.0

**Pull request closed without merging**